scripts/bootstrap/buildbootstrapcli.sh

```diff
@@ -0,0 +1,271 @@
+    case $lowerI in
+    -h|--help)
+        usage
+        exit 1
+        ;;
+    -arch)
+        shift
+        __build_arch=$1
+        ;;
+    -rid)
+        shift
+        __runtime_id=$1
+        ;;
+    -os)
+        shift
+        __build_os=$1
+        ;;
+    -debug)
+        __configuration=debug
+        ;;
+    -release)
+        __configuration=release
+        ;;
+    -corelib)
+        shift
+        __corelib=$1
+        ;;
+    -seedcli)
+        shift
+        __seedclipath=$1
+        ;;
+    -clang)
+        shift
+        __clangversion=clang$1
+        ;;
+     *)
+    echo "Unknown argument to build.sh $1"; exit 1
+    esac
+    shift
+done
+
+# NOTE: when realpath is not present, use readlink -e or readlink --canonicalize depending on the platform
+__seedclipath=`realpath $__seedclipath`
+
+mkdir -p $__runtime_id-$__build_arch
+cd $__runtime_id-$__build_arch
+
+if [[ -d dotnetcli ]]; then
+    /bin/rm -r dotnetcli
+fi
+mkdir -p dotnetcli
+cp -r $__seedclipath/* dotnetcli
+
+__frameworkversion="2.0.0"
+__sdkversion="2.0.0"
+__fxrversion="2.0.0"
+
+echo "**** DETECTING VERSIONS IN SEED CLI ****"
+
+__frameworkversion=`get_max_version dotnetcli/shared/Microsoft.NETCore.App`
+__sdkversion=`get_max_version dotnetcli/sdk`
+__fxrversion=`get_max_version dotnetcli/host/fxr`
+
+echo "Framework version: $__frameworkversion"
+echo "SDK version:       $__sdkversion"
+echo "FXR version:       $__fxrversion"
+
+__frameworkpath="dotnetcli/shared/Microsoft.NETCore.App/$__frameworkversion"
+
+echo "**** DETECTING GIT COMMIT HASHES ****"
+
+# Extract the git commit hashes representig the state of the three repos that
+# the seed cli package was built from
+__coreclrhash=`strings $__seedclipath/shared/Microsoft.NETCore.App/$__frameworkversion/libcoreclr.so | grep "@(#)" | grep -o "[a-f0-9]\{40\}"`
+__corefxhash=`strings $__seedclipath/shared/Microsoft.NETCore.App/$__frameworkversion/System.Native.so | grep "@(#)" | grep -o "[a-f0-9]\{40\}"`
+__coresetuphash=`strings $__seedclipath/dotnet | grep "[a-f0-9]\{40\}"`
+
+echo "coreclr hash:    $__coreclrhash"
+echo "corefx hash:     $__corefxhash"
+echo "core-setup hash: $__coresetuphash"
+
+# Clone the three repos if they were not clonned yet. If the folders already
+# exist, leave them alone. This allows patching the cloned sources as needed
+
+if [[ ! -d coreclr ]]; then
+    echo "**** CLONING CORECLR REPOSITORY ****"
+    git clone https://github.com/dotnet/coreclr.git
+    cd coreclr
+    git checkout $__coreclrhash
+    cd ..
+fi
+
+if [[ ! -d corefx ]]; then
+    echo "**** CLONING COREFX REPOSITORY ****"
+    git clone https://github.com/dotnet/corefx.git
+    cd  corefx
+    git checkout $__corefxhash
+    cd ..
+fi
+
+if [[ ! -d core-setup ]]; then
+    echo "**** CLONING CORE-SETUP REPOSITORY ****"
+    git clone https://github.com/dotnet/core-setup.git
+    cd  core-setup
+    git checkout $__coresetuphash
+    cd ..
+fi
+
+echo "**** BUILDING CORE-SETUP NATIVE COMPONENTS ****"
```

Comment:
Is there a reason that core-setup is built first?

Reply:
None in particular. The repos can be built in any arbitrary order. This order came from an original script which I used as a basis for this one.